### PR TITLE
Update games-simulation/openttd and dependencies

### PR DESCRIPTION
**Pull request by winterheart**
Rewrite games-simulation/openttd for cmake.eclass, update other dependencies.

### CODE REVIEW

File: games-util/grfcodec/grfcodec-6.0.6_p20210310-r1.ebuild [hunk 1]
```diff
@@ -8,18 +8,15 @@ inherit cmake
...
 SLOT="0"
 KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
-IUSE=""
 
 RDEPEND="media-libs/libpng:0"
-DEPEND="
-	${RDEPEND}
-	dev-lang/perl
+DEPEND="${RDEPEND}
 	dev-libs/boost
```
Comment (a17r) [inline games-util/grfcodec/grfcodec-6.0.6_p20210310-r1.ebuild:20]: This change does not warrant a revbump, you can do it in place.

Re commit message: `games-util/grfcodec: update ebuild`
I'd say `update ebuild` is implied by any way we touch an ebuild.

Suggestion:
`games-util/grfcodec: drop unused DEPEND, fix HOMEPAGE`

File: games-simulation/openttd/openttd-1.11.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+inherit cmake
+
+DESCRIPTION="OpenTTD is a clone of Transport Tycoon Deluxe"
+HOMEPAGE="https://www.openttd.org/"
+SRC_URI="https://cdn.openttd.org/openttd-releases/${PV}/${P}-source.tar.xz"
+
+KEYWORDS="~amd64 ~ppc64 ~x86"
+LICENSE="GPL-2"
+SLOT="0"
```
Comment (a17r) [inline games-simulation/openttd/openttd-1.11.2.ebuild:14]: `KEYWORDS` should be below `SLOT`

File: games-simulation/openttd/openttd-1.11.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="OpenTTD is a clone of Transport Tycoon Deluxe"
```
Comment (a17r) [inline games-simulation/openttd/openttd-1.11.2.ebuild:8]: `PN` should not be repeated in `DESCRIPTION`.

File: games-simulation/openttd/openttd-1.11.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		)
+	)
+	openmedia? ( >=games-misc/opengfx-0.6.1 )
+	timidity? ( media-sound/timidity++ )"
+
+DOCS=( docs/directory_structure.md )
+
+src_prepare() {
+	sed -i -e "/COMMAND gzip -9 -n -f/d" -e "s/.gz//" \
+		cmake/InstallAndPackage.cmake || die "sed failed"
```
Comment (a17r) [inline games-simulation/openttd/openttd-1.11.2.ebuild:57]: `sed` will fail silently. Use a patch instead to detect any upstream changes on a future version bump.

File: games-simulation/openttd/openttd-1.11.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_BINDIR=bin
+		-DCMAKE_INSTALL_DATADIR=share
```
Comment (a17r) [inline games-simulation/openttd/openttd-1.11.2.ebuild:64]: That looks like a weird workaround for wrong upstream use of `GNUInstallDirs` module.

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 6]
```diff
@@ -55,7 +54,7 @@ src_install() {
 		DO_NOT_INSTALL_LICENSE="true"
 		DO_NOT_INSTALL_CHANGELOG="true"
 		GIMP=""
-		INSTALL_DIR="${ED}/usr/share/games/openttd/data/"
+		INSTALL_DIR="${ED}/usr/share/openttd/baseset/"
 		PYHTON="${EPYTHON}"
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild:58]: typo?

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+DOCS=( "README.md" "changelog.txt" )
+
+src_prepare() {
+	default
+
+	python-any-r1_pkg_setup
+}
+
+src_compile() {
+	myemakeargs=(
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild]: `local myemakeargs`

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+		GIMP=""
+		PYHTON="${EPYTHON}"
+	)
+
+	emake "${myemakeargs[@]}" all
+}
+
+src_test() {
+	myemakeargs=(
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild]: `local myemakeargs`

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+src_prepare() {
+	default
+
+	python-any-r1_pkg_setup
+}
+
+src_compile() {
+	myemakeargs=(
+		GIMP=""
+		PYHTON="${EPYTHON}"
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild]: typo?

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+	)
+
+	emake "${myemakeargs[@]}" all
+}
+
+src_test() {
+	myemakeargs=(
+		GIMP=""
+		PYHTON="${EPYTHON}"
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild]: typo?

File: games-misc/opengfx/opengfx-0.6.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+	)
+
+	emake "${myemakeargs[@]}" check
+}
+
+src_install() {
+	myemakeargs=(
```
Comment (a17r) [inline games-misc/opengfx/opengfx-0.6.1.ebuild]: `local myemakeargs`

**Review by a17r — CHANGES_REQUESTED** [review]

File: games-simulation/openttd/openttd-1.11.2.ebuild (hunk 5 above)
Comment (winterheart) [reply games-simulation/openttd/openttd-1.11.2.ebuild:61]: Yes, it is. If not define these options, package will install binaries into `/usr/games` and data into `/usr/shage/games`.

File: games-misc/opengfx/opengfx-0.6.1.ebuild (hunk 9 above)
Comment (winterheart) [reply games-misc/opengfx/opengfx-0.6.1.ebuild]: Huh, that funny. It comes from already existing 0.6.0. Fixed now.

**Comment by winterheart** [comment]
Fixed.

**Review by a17r — APPROVED** [review]
lgtm purely from reading the ebuilds.

**Comment by chewi** [comment]
Great work. I've given it a try and it looks good. Just one thing, any particular reason you dropped the `debug` flag and the live ebuild?

**Comment by winterheart** [comment]
I don't see any purposes for debug and live version for game package. 9999 is broken all time and needs extensive maintenance.

**Pull request closed without merging**